pkg/config/remote/service/service.go

```diff
@@ -0,0 +1,516 @@
+//go:generate go run github.com/shuLhan/go-bindata/cmd/go-bindata -pkg meta -prefix meta/ -include .* -o ./meta/meta.go meta/
+
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package service
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/json"
+	"fmt"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/service/tuf"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/store"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+)
+
+const (
+	defaultRefreshInterval = time.Second * 5
```

Comment:
We probably want to start higher for this value with this first version. It should be mostly transparent to customers if we decide to lower it in the future. The other way around would be problematic as customers would have no incentive to change.
I also think we should make the value configurable with a hard minimum of something between 1-5s.

What do you think of 1 min for this release ?

Reply:
Works for me. I changed the default to 60 seconds, made it configurable and set a minimum